src/player.cpp

```diff
@@ -6147,7 +6189,9 @@ void player::process_effects() {
 
             // Handle fatigue
             val = it.get_mod("FATIGUE", reduced);
-            if (val != 0) {
+            // Prevent ongoing fatigue effects while asleep.
+            // These are meant to change how fast you get tired, not how long you sleep.
+            if (val != 0 && !in_sleep_state()) {
```

Comment:
I for one think that the cough wakeup should be part of diseases, but pretty sure it can be toned down a bit instead.